src/components/BrandConfigProvider/styles/createStyles.ts

```diff
@@ -14,16 +14,19 @@ import { Theme } from '../theme/ThemeProvider';
  */
 export const createStyles: CreateStyles = (namespace, styleBuilder) => {
   const defaultStyles = (theme: Theme) => {
-    return [namespace, styleBuilder(theme)] as any;
+    return [namespace, styleBuilder(theme)] as const;
   };
 
   return defaultStyles;
 };
 
-export type CreateStyles = <N extends string, T extends NamedStyles<T>>(
+export type CreateStyles<Names = string> = <
```

Comment:
Allow providing a type to say what names are valid. By default accept any string.